skymp5-server/cpp/server_guest_lib/MpActor.cpp

```
@@ -365,6 +365,40 @@ void MpActor::MpApiDeath(MpActor* killer)
   }
 }
 
+void MpActor::ModifyActorValue(espm::ActorValue av, float value)
+{
+  if ((av == espm::ActorValue::Health || av == espm::ActorValue::Stamina ||
+       av == espm::ActorValue::Magicka) == false) {
+    return;
+  }
+  BaseActorValues base =
+    GetBaseActorValues(GetParent(), GetBaseId(), GetRaceId());
+  MpChangeForm form = GetChangeForm();
+  float hp = form.healthPercentage;
+  float mp = form.magickaPercentage;
+  float sp = form.staminaPercentage;
+  switch (av) {
+    case espm::ActorValue::Health:
+      hp = form.healthPercentage + value / base.health;
+      // its temporal
+      hp = hp < 0 ? 0 : hp > 1 ? 1 : hp;
+      break;
+    case espm::ActorValue::Stamina:
+      sp = form.staminaPercentage + value / base.stamina;
+      // its temporal
+      sp = sp < 0 ? 0 : sp > 1 ? 1 : sp;
+      break;
+    case espm::ActorValue::Magicka:
+      mp = form.magickaPercentage + value / base.magicka;
+      // its temporal
+      mp = mp < 0 ? 0 : mp > 1 ? 1 : mp;
+      break;
+    default:
+      return;
```

we should throw `std::runtime_error("Unsupported actor value 123")`